### ISSUE TITLE
mu4e-git compatibility broken

### ISSUE THREAD

**Issue by innerout** [agpchil/mu4e-maildirs-extension#52]
Something has changed in the mu4e-git version and when mu4e-maildirs-extension runs along mu4e  mu4e-main view becomes empty.

**Comment by eigengrau** [comment]
I just upgraded mu4e to 2fd80584. mu4e’s main-view looks okay to me, it’s just that the maildirs list is not rendered:

```
Error (use-package): mu4e-maildirs-extension/:catch: Symbol’s value as variable is void: mu4e~main-buffer-name
```

It looks like that variable is public API now (`mu4e-main-buffer-name`). After changing the variable reference accordingly, opening mu4e’s main view indeed results in an empty buffer. In the message buffer I see:

```
Search failed: "
  Misc"
```

Toggling edebug on `mu4e-maildirs-extension-update` shows that the mu4e main view buffer is empty when the function runs. Possibly `mu4e-main-mode-hook` was changed and now runs immediately upon entering the mode, before anything is inserted into the buffer.

As a temporary workaround, it’s possible to set `mu4e-maildirs-extension-insert-before-str` to the empty string.
